base/namedtuple.jl

```diff
@@ -209,6 +209,30 @@ function map(f, nt::NamedTuple{names}, nts::NamedTuple...) where names
     NamedTuple{names}(map(f, map(Tuple, (nt, nts...))...))
 end
 
+function _foldl_impl(
+    op,
+    init,
+    itr::Iterators.Pairs{<:Any,<:Any,<:Any,<:NamedTuple{names}},
+) where {names}
+    if @generated
+        ex = :init
+        for (i, n) in enumerate(names)
+            ex = :(op($ex, $(QuoteNode(n)) => vals[$i]))
```

Comment:
@JeffBezanson Is it okay for the `Iterators.Pairs` iterator to be indexed directly by `names` in this order? I thought you complained something about that recently to me.